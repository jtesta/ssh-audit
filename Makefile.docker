VERSION = $(shell grep VERSION src/ssh_audit/globals.py | grep -E -o "'(v.*)'" | tr -d "'")
ifeq ($(VERSION),)
  $(error "could not determine version!")
endif

all:
	docker build -t positronsecurity/ssh-audit:${VERSION} .

upload:
	docker login
	docker buildx build \
		--platform linux/amd64,linux/arm64,linux/arm/v7 \
		--tag positronsecurity/ssh-audit:${VERSION} \
		--tag positronsecurity/ssh-audit:latest \
		--push \
		.
