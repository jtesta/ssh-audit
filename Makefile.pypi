all:
	cp ../ssh-audit.py sshaudit/sshaudit.py
	cp ../LICENSE sshaudit/LICENSE
	cp ../README.md sshaudit/README.md
	python3 setup.py sdist bdist_wheel

uploadtest:
	twine upload --repository-url https://test.pypi.org/legacy/ dist/*

uploadprod:
	twine upload dist/*

clean:
	rm -rf parts/ prime/ snap/ stage/ build/ dist/ *.egg-info/ sshaudit/sshaudit.py sshaudit/LICENSE sshaudit/README.md ssh-audit*.snap
