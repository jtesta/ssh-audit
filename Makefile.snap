all:
	cp ../ssh-audit.py sshaudit/sshaudit.py
	cp ../README.md sshaudit/README.md
	echo -e "\n\nDid you remember to bump the version number in snapcraft.yaml?\n\n"
	snapcraft --use-lxd

clean:
	rm -rf parts/ prime/ snap/ stage/ build/ dist/ *.egg-info/ sshaudit/sshaudit.py sshaudit/LICENSE sshaudit/README.md ssh-audit*.snap
